latex_engine_cmds.bzl

```diff
@@ -0,0 +1,258 @@
+"""
+This module contains the specializations of commands depending on tool or engine.
+"""
+
+def to_dirname(inp):
+    return inp.dirname
+
+def get_common_args(ctx, dep_tools, files):
+    """
+    Set up environment variables common for all commands.
+
+    Latex and a set of scripts and binaries in the tool suite
+    makes use of a library, kpathsea.
+    In general latex distributions are encouraged to follow the 'TDS'
+    structure. And tools and script might make assumptions that the
+    layout of directories respects that structure.
+
+    But from our perspective, trying to shoehorn the partioned repositories
+    according
+    (E.g. <sandbox>/bazel-latex/external/<ext_texlive_repo>/<files>)
+    to the TDS creates an unnecessary complexity.
+    Also kpathsea tries to be efficient about looking up files.
+    So to derive if a folder is of interest is check if the number
```

Comment:
This line is a bit unclear - do you mean `So to derive if a folder is of interest, we check...` maybe? The `is check` seems wrong, but I'm not sure what it should actually be.

Reply:
Yeah, I noticed that the sentence was weird after I had pushed it. I'll fix it!

Reply:
done